src/Entity/Policy.php

```diff
@@ -0,0 +1,149 @@
+<?php
+
+namespace Drupal\password_enhancements\Entity;
+
+use Drupal\Core\Config\Entity\ConfigEntityBase;
+
+/**
+ * Defines the Password Constraint config entity.
+ *
+ * @ConfigEntityType(
+ *   id = "password_enhancements_policy",
+ *   label = @Translation("Password policy"),
+ *   label_collection = @Translation("Password policies"),
+ *   handlers = {
+ *     "storage" = "Drupal\password_enhancements\Entity\Storage\PolicyEntityStorage",
+ *     "route_provider" = {
+ *       "html" = "Drupal\Core\Entity\Routing\DefaultHtmlRouteProvider",
+ *     },
+ *     "form" = {
+ *       "default" = "Drupal\password_enhancements\Form\PolicyForm",
+ *       "delete" = "Drupal\password_enhancements\Form\PolicyDeleteForm",
+ *     },
+ *     "list_builder" = "Drupal\password_enhancements\Entity\PolicyListBuilder",
+ *   },
+ *   config_prefix = "policy",
+ *   admin_permission = "administer user password settings",
+ *   entity_keys = {
+ *     "id" = "role",
+ *     "label" = "role",
+ *   },
+ *   config_export = {
+ *     "expireDays",
+ *     "minimumRequiredConstraints",
+ *     "status",
+ *     "priority",
+ *     "role",
+ *   },
+ *   links = {
+ *     "collection" = "/admin/config/people/password/policies",
+ *     "add-form" = "/admin/config/people/password/policy/add",
+ *     "edit-form" = "/admin/config/people/password/policy/{password_enhancements_policy}",
+ *     "delete-form" = "/admin/config/people/password/policy/{password_enhancements_policy}/delete",
+ *   }
+ * )
+ */
+class Policy extends ConfigEntityBase implements PolicyInterface {
+
+  /**
+   * Expiry of the password in days, 0 means that passwords doesn't expire.
+   *
+   * @var int
+   */
+  protected $expireDays = 0;
+
+  /**
+   * The number of constraints that are required.
+   *
+   * @var int
+   */
+  protected $minimumRequiredConstraints = 1;
+
+  /**
+   * The priority of the policy.
+   *
+   * @var int
+   */
+  protected $priority = 1;
+
+  /**
+   * Role ID.
+   *
+   * @var string
+   */
+  protected $role = '';
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(array $values, $entity_type) {
+    // The entity builder does not recognize the role configured as ID in some
```

Comment:
Since it is possible to acquire the ID through a direct call to the property I'd rather leave it like this instead of overriding the id getter method, feels more safe.

Reply:
This is what I suggested I guess in call, that you should override what the `id()` method returns on an entity. You do not need to return the $id property (you do not need that property) if your entity's primary id is what it is stored in the role property. 

```php
<?php
  /**
   * {@inheritdoc}
   */
  public function id() {
    return isset($this->id) ? $this->id : NULL;
  }
```